### ISSUE TITLE
support for nuxt generate

### ISSUE THREAD

**Issue by mateuszk969** [reallifedigital/nuxt-image-loader-module#1]
Hello,
is there any way to make this module work with nuxt generate(and deplying to static hosting)? 

Regards,
Mateusz Kurzempa

**Comment by barryrld** [comment]
Hi @mateuszk969 

I haven't tested with nuxt generate just yet. This module is in early stages, having only released it last week.

I'll look to see whether there is a way to get this working so that when the static generation is happening that the images can be pre-generated also.

Like I said, I haven't tried it just yet but I appreciate your feedback as it was something that was on mind to address.

Have you tried running nuxt generate? If so, what was your experience? Did you see any errors?

Thanks,

Barry

**Comment by mateuszk969** [comment]
Hello, yes i have tried it and sadly it doesn't work -  i am trying to get it working since yesterday :) 
The error i getis 404, because images are generated using middleware and middleware is only working when nuxt server is up. to make it work with generate path shouldn't consist of ? because then we get error that "?" char is not allowed. 
The solution would be to change naming convention(?) to lets say /static/small/imgName  

Do you have idea how to make copies of those files during building phase?

**Comment by dev7ch** [comment]
@barryrld  @mateuszk969 i am using this module in combination with netlify cms withgenerate`.

It´s might be kind of a hack, but in static folder there is `images-styles`, which can be used.

```
      if (this.config.attributes.gallery.list) {
        for (let i = 0; i < this.config.attributes.gallery.list.length; i++) {
          let img = this.config.attributes.gallery.list[i].image.src;
          let filename = img.substring(img.lastIndexOf('/')+1);
          let thumb;

          if (process.browser) {
            img = '/image-styles/uploads/' + filename.substring(0, filename.lastIndexOf(".")) + "--full" + filename.substring(filename.lastIndexOf("."));
            thumb = '/image-styles/uploads/' + filename.substring(0, filename.lastIndexOf(".")) + "--small" + filename.substring(filename.lastIndexOf("."));
          } else {
            img = this.config.attributes.gallery.list[i].image.src + '?style=full';
            thumb = this.config.attributes.gallery.list[i].image.src +'?style=small'
          }

          this.items.push({
            src: img,
            thumbnail: thumb,
            w: '1024',
            h: '640',
            //test: this.config.attributes.gallery.list[i].image.src.naturalHeight
          })
        }
      }

      if (this.config.attributes.cover) {
        let cover = this.config.attributes.cover;
        let coverName = cover.substring(cover.lastIndexOf('/')+1);
        let coverImg = coverName.substring(0, coverName.lastIndexOf(".")) + "--medium" + coverName.substring(coverName.lastIndexOf("."));

        if (process.browser) {
          this.coverImage = '/image-styles/uploads/' + coverImg;
        } else  {
          this.coverImage = this.config.attributes.cover + '?style=medium'
        }
      }
    },
```

(only tested 30min, not 100% ready)

**Comment by dev7ch** [comment]
@mateuszk969 [Here on this Repo](https://github.com/eflorido/eloisa-florido-navarro.com/blob/master/nuxt.config.js) you find an example how you could generate *.json map files from

**Comment by barryrld** [comment]
I'm working on a more integrative (non hacky) solution over in the generate-support branch. It's working reasonably well as of last night, however I still need to do a bit of work on it and write the docs/tests. This branch will likely become version 1.1 as it shouldn't break anything from before, although it will add a `nuxt-img` component to allow for dynamic rewriting of URLs for generate support.

**Comment by barryrld** [comment]
Update: I've released a new version 1.1.0 of this module which should still work for backwards compatibility, although there is a new `<nuxt-img />` component available for supporting static generation.

Please try this latest version to see how you get on. I've also updated the docs, so please see a full explanation there.

There is no longer a requirement to run the `build` or `dev` steps before `generate` to artificially generate the static images. `nuxt generate` now takes care of every renderable image in your app because it collects image calls when rendering the `<nuxt-img />` component and then rewrite the `src` attribute _relative_ to the route of the page.

I've still got to write the test cases across the board as I've been a bit limited on time.

@mateuszk969 - Thanks for your input. The latest version follows fairly closely to your suggestion of removing the query strings in favor of a rewritten relative path to the generated file.

@dev7ch - Thanks for your suggestions both in this issue and also the comment you made in the feature branch. Unfortunately, I wasn't able to completely follow along with the implementation of your use case. You mentioned an image slider or something similar. Are you able to open a separate issue and provide a bit of extra background and code to help debug your issue? That being said, please try with the latest version.

Cheers!

**Comment by barryrld** [comment]
Marking this as closed as fixed with the [v1.1.0](https://github.com/reallifedigital/nuxt-image-loader-module/releases/tag/v1.1.0) release.

If you have a specific issue implementing this module with `nuxt generate`, please open a new issue.

Thanks to @mateuszk969 and @dev7ch in prompting the development work required to get generated images working.

**Comment by dev7ch** [comment]
@barryrId thanks for report. Next week i ll be back to the project and will report the usage of your module.
